plugin.rb

```diff
@@ -183,49 +164,10 @@ def self.skip_db?
     ].each { |klass| klass.include(DiscourseSolved::TopicAnswerMixin) }
   end
 
-  # we got to do a one time upgrade
-  if !::DiscourseSolved.skip_db?
-    unless Discourse.redis.get("solved_already_upgraded")
-      unless UserAction.where(action_type: UserAction::SOLVED).exists?
-        Rails.logger.info("Upgrading storage for solved")
-        sql = <<~SQL
-          INSERT INTO user_actions(action_type,
-                                   user_id,
-                                   target_topic_id,
-                                   target_post_id,
-                                   acting_user_id,
-                                   created_at,
-                                   updated_at)
-          SELECT :solved,
-                 p.user_id,
-                 p.topic_id,
-                 p.id,
-                 t.user_id,
-                 pc.created_at,
-                 pc.updated_at
-          FROM
-            post_custom_fields pc
-          JOIN
-            posts p ON p.id = pc.post_id
-          JOIN
-            topics t ON t.id = p.topic_id
-          WHERE
-            pc.name = 'is_accepted_answer' AND
-            pc.value = 'true' AND
-            p.user_id IS NOT NULL
-        SQL
-
-        DB.exec(sql, solved: UserAction::SOLVED)
-      end
-      Discourse.redis.set("solved_already_upgraded", "true")
-    end
-  end
-
-  topic_view_post_custom_fields_allowlister { [::DiscourseSolved::IS_ACCEPTED_ANSWER_CUSTOM_FIELD] }
-  TopicList.preloaded_custom_fields << ::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD
+  register_category_list_topics_preloader_associations(:solved) if SiteSetting.solved_enabled
```

Comment:
This API is not yet available on the `stable` branch so we need to update the plugins compatibility file.